agent/main.go

```diff
@@ -90,6 +92,33 @@ func setupflags() {
 	for _, hiddenFlag := range hiddenFlags {
 		_ = pflag.CommandLine.MarkHidden(hiddenFlag)
 	}
+	feature.MutableGates.AddFlag(pflag.CommandLine)
+}
+
+func handleHostRegistration(k8sClient client.Client, hostName string, logger logr.Logger) (err error) {
+	if feature.Gates.Enabled(feature.SecureAccess) {
+		logger.Info("secure access enabled, waiting for host to be registered by ByoAdmission Controller")
+	} else {
+		registration.LocalHostRegistrar = &registration.HostRegistrar{K8sClient: k8sClient}
+		err := registration.LocalHostRegistrar.Register(hostName, namespace, labels)
+		return err
+	}
+	return nil
+}
+
+func setupTemplateParser() *cloudinit.TemplateParser {
+	var templateParser *cloudinit.TemplateParser
+	if feature.Gates.Enabled(feature.SecureAccess) {
+		templateParser = nil
+	} else {
+		templateParser = &cloudinit.TemplateParser{
+			Template: registration.HostInfo{
+				DefaultNetworkInterfaceName: registration.LocalHostRegistrar.ByoHostInfo.DefaultNetworkInterfaceName,
+			},
+		}
+	}
+
+	return templateParser
```

Comment:
Question: `HostReconciler` will get called only in case of some Create/Update/Delete event of `ByoHost`. I am not sure if we require some handling for template parser.

Reply:
The `templateParser` is part of the `HostReconciler` bootstrap process which in turn is used to bootstrap the node. The `setupTemplateParser` sets up a minimal `reconciler` which will get updated once the `byohost` is registered.

Reply:
> The setupTemplateParser sets up a minimal reconciler which will get updated once the byohost is registered

What's the limitation if we create the `HostReconciler` the same way as of today?

Reply:
As discussed [here](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/issues/428#issuecomment-1067645100), the `HostReconciler` has some dependencies like network interface setup, with the host registration process. Since we are skipping the registration, the `HostReconciler` will have to be updated.